pkg/controller/backup_sync_controller.go

```diff
@@ -134,6 +136,35 @@ func (c *backupSyncController) run() {
 
 	for _, location := range locations {
 		log := c.logger.WithField("backupLocation", location.Name)
+
+		syncPeriod := c.defaultBackupSyncPeriod
+		if location.Spec.BackupSyncPeriod != nil {
+			syncPeriod = location.Spec.BackupSyncPeriod.Duration
+			if syncPeriod == 0 {
+				log.Debug("Backup sync period for this location is set to 0, skipping sync")
+				continue
+			}
+
+			if syncPeriod < 0 {
+				log.Debug("Backup sync period must be non-negative")
+				syncPeriod = c.defaultBackupSyncPeriod
+			}
+		}
+
+		lastSync := location.Status.LastSyncedTime
+		if lastSync != nil {
+			log.Debug("Checking if backups need to be synced at this time for this location")
+			nextSync := lastSync.Add(syncPeriod)
+			if time.Now().UTC().Before(nextSync) {
+				c.logger.WithFields(logrus.Fields{
+					"backupLocation":   location.Name,
+					"backupSyncPeriod": syncPeriod,
+					"lastSync":         lastSync,
+					"nextSync":         nextSync}).Debug("Sync not required at this time")
```

Comment:
Is this log necessary? If the sync was configured to be every hour, then this would be printed every minute until the hour since the controller sync is every minute, right?

Reply:
You're right, this will be too much logging. I removed the logs.